### PR TITLE
uuid generation

### PR DESCRIPTION
**Pull request by tanvi029**

### CODE REVIEW

**Review by kailash — APPROVED** [review]

**Comment by kailash** [comment]
ok to test

File: src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java [hunk 1]
```diff
@@ -155,57 +160,125 @@ public ValidatorService validateItem(JsonObject request,
...
+                    } else {
+                      LOGGER.debug("Fail: resource group doesn't exist");
+                      handler.handle(Future.failedFuture(VALIDATION_FAILURE_MSG));
+                    }
+                  });
+            } else {
+              LOGGER.debug("Fail: Provider doesn't exist");
+              handler.handle(Future.failedFuture(VALIDATION_FAILURE_MSG));
+            }
+          });
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java:212]: Can be squashed into one query to search both provider and RG existence

File: src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java [hunk 2]
```diff
@@ -155,57 +160,125 @@ public ValidatorService validateItem(JsonObject request,
...
+
+        if (!isValidUuid(id)) {
+          handler.handle(Future.failedFuture("validation failed. Incorrect id"));
+        }
+      } else if (!isUACinstance && !request.containsKey("id")) {
+        UUID uuid = UUID.randomUUID();
+        request.put(ID, uuid.toString());
+      } else if (isUACinstance && !request.containsKey("id")) {
+        handler.handle(Future.failedFuture("id not found"));
+      }
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java:228]: Same as code at lines 165-176

File: src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java [hunk 3]
```diff
@@ -155,57 +160,125 @@ public ValidatorService validateItem(JsonObject request,
...
+        if (!isValidUuid(id)) {
+          handler.handle(Future.failedFuture("validation failed. Incorrect id"));
+        }
+      } else if (!isUACinstance && !request.containsKey("id")) {
+        byte[] inputBytes = request.getString("name").getBytes(StandardCharsets.UTF_8);
+        UUID uuid = UUID.nameUUIDFromBytes(inputBytes);
+        request.put("id", uuid.toString());
+      } else if (isUACinstance && !request.containsKey("id")) {
+        handler.handle(Future.failedFuture("id not found"));
+      }
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java:261]: same as code at lines 165-176

File: src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java [hunk 4]
```diff
@@ -155,57 +160,125 @@ public ValidatorService validateItem(JsonObject request,
...
+        if (!isValidUuid(id)) {
+          handler.handle(Future.failedFuture("validation failed. Incorrect id"));
+        }
+      } else if (!isUACinstance && !request.containsKey("id")) {
+        UUID uuid = UUID.randomUUID();
+        request.put("id", uuid.toString());
+        LOGGER.debug("Info: id generated: " + uuid.toString());
+      } else if (isUACinstance && !request.containsKey("id")) {
+        handler.handle(Future.failedFuture("id not found"));
+      }
```
Comment (pranavrd) [inline src/main/java/iudx/catalogue/server/validator/ValidatorServiceImpl.java:279]: same as code at lines 165-176

**Comment by pranavrd** [comment]
retest this please

**Review by pranavrd — APPROVED** [review]

**Pull request merged**